Tasks/PackerBuild/src/taskParameters.ts

```diff
@@ -104,4 +110,16 @@ export default class TaskParameters {
 
         return inputPath;
     }
+
+    private _getAzureADGraphCredentials(connectedService: string): msRestAzure.ApplicationTokenCredentials {
+        var servicePrincipalId: string = tl.getEndpointAuthorizationParameter(connectedService, "serviceprincipalid", false);
+        var servicePrincipalKey: string = tl.getEndpointAuthorizationParameter(connectedService, "serviceprincipalkey", false);
+        var tenantId: string = tl.getEndpointAuthorizationParameter(connectedService, "tenantid", false);
+        var envAuthorityUrl: string = tl.getEndpointDataParameter(connectedService, 'environmentauthorityurl', false);
+        envAuthorityUrl = (envAuthorityUrl != null) ? envAuthorityUrl : "https://login.windows.net/";
+        var activeDirectoryResourceId: string = tl.getEndpointDataParameter(connectedService, 'graphUrl', false);
+        activeDirectoryResourceId = (activeDirectoryResourceId != null) ? activeDirectoryResourceId : "https://graph.windows.net/";
+        var credentials = new msRestAzure.ApplicationTokenCredentials(servicePrincipalId, tenantId, servicePrincipalKey, null, envAuthorityUrl, activeDirectoryResourceId, false);
```

Comment:
consider creating credentials with the graph resource URL. rename the field in the credentials class also